# 01 - Lectura y Discovery

**Objetivo de esta notebook:** abrir los datos por primera vez y conocerlos.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("Librerías cargadas OK")



Librerías cargadas OK


In [2]:
train = pd.read_excel("../data/raw/smoking_train_labeled.xlsx")
#convertilos en dataset
test  = pd.read_excel("../data/raw/smoking_unlabeled.xlsx")

print("train ( este para entrenar):", train.shape)
print("test  (este para predecir):", test.shape)


train ( este para entrenar): (50000, 27)
test  (este para predecir): (5692, 26)


In [3]:
train.head(10)
#muestra las 10 filas

,ID,gender,age,height(cm),weight(kg),waist(cm),eyesight(left),eyesight(right),hearing(left),hearing(right),...,hemoglobin,Urine protein,serum creatinine,AST,ALT,Gtp,oral,dental caries,tartar,smoking
0,0,F,40,155,60,3.377083,0.043056,0.041667,0.041667,0.041667,...,0.506250,0.041667,0.004861,0.750000,0.791667,1.125000,Y,0,Y,0
1,1,F,40,160,60,3.375000,0.005556,0.004167,0.041667,0.041667,...,0.504861,0.041667,0.004167,0.916667,0.791667,0.750000,Y,0,Y,0
2,2,M,55,170,60,3.333333,0.005556,0.005556,0.041667,0.041667,...,0.630556,0.041667,0.041667,0.875000,0.666667,0.916667,Y,0,N,1
3,3,M,40,165,70,3.666667,0.045139,0.045139,0.041667,0.041667,...,0.588194,0.041667,0.041667,0.791667,1.083333,0.750000,Y,0,Y,0
4,4,F,40,155,60,3.583333,0.041667,0.041667,0.041667,0.041667,...,0.503472,0.041667,0.004167,0.666667,0.583333,0.916667,Y,0,N,0
5,5,M,30,180,75,3.541667,0.043056,0.043056,0.041667,0.041667,...,0.668056,0.041667,0.043056,0.750000,1.125000,1.375000,Y,0,Y,0
6,6,M,40,160,60,3.545139,0.041667,0.041667,0.041667,0.041667,...,0.708333,0.041667,0.004861,0.875000,1.125000,1.625000,Y,1,Y,1
7,7,M,45,165,90,4.000000,0.043056,0.041667,0.041667,0.041667,...,0.625000,0.041667,0.043750,1.583333,2.958333,4.625000,Y,0,Y,0
8,9,F,50,150,60,3.541667,0.004861,0.005556,0.041667,0.041667,...,0.546528,0.041667,0.005556,1.291667,1.291667,0.583333,Y,0,N,0
9,10,M,45,175,75,3.708333,0.041667,0.041667,0.041667,0.041667,...,0.666667,0.041667,0.005556,1.083333,1.000000,2.625000,Y,0,N,0


In [4]:
# ver tipo de dato de cada columna y cuántos valores que no sean nulos tiene
train.info()


<class 'pandas.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 27 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ID                   50000 non-null  int64  
 1   gender               50000 non-null  str    
 2   age                  50000 non-null  int64  
 3   height(cm)           50000 non-null  int64  
 4   weight(kg)           50000 non-null  int64  
 5   waist(cm)            50000 non-null  float64
 6   eyesight(left)       50000 non-null  float64
 7   eyesight(right)      50000 non-null  float64
 8   hearing(left)        50000 non-null  float64
 9   hearing(right)       50000 non-null  float64
 10  systolic             50000 non-null  float64
 11  relaxation           50000 non-null  float64
 12  fasting blood sugar  50000 non-null  float64
 13  Cholesterol          50000 non-null  float64
 14  triglyceride         50000 non-null  float64
 15  HDL                  50000 non-null  float64
 1

In [5]:
#valores raros
train.describe()


,ID,age,height(cm),weight(kg),waist(cm),eyesight(left),eyesight(right),hearing(left),hearing(right),systolic,...,HDL,LDL,hemoglobin,Urine protein,serum creatinine,AST,ALT,Gtp,dental caries,smoking
count,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,...,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000
mean,26284.994580,44.189300,164.652000,65.878700,3.414852,0.028446,0.028304,0.042736,0.042758,5.062531,...,2.387543,4.790698,0.593785,0.045290,0.018821,1.093193,1.129345,1.668142,0.213120,0.366580
std,16015.070567,12.090086,9.208778,12.831355,0.386765,0.024007,0.024216,0.006588,0.006655,0.570100,...,0.615559,1.709238,0.065870,0.016872,0.018432,0.827986,1.327431,2.110347,0.409516,0.481875
min,0.000000,20.000000,130.000000,30.000000,2.125000,0.000694,0.000694,0.041667,0.041667,2.958333,...,0.166667,0.041667,0.172917,0.041667,0.000694,0.250000,0.041667,0.041667,0.000000,0.000000
25%,12499.750000,40.000000,160.000000,55.000000,3.166667,0.005556,0.005556,0.041667,0.041667,4.666667,...,1.958333,3.833333,0.545833,0.041667,0.005556,0.791667,0.625000,0.708333,0.000000,0.000000
50%,24999.500000,40.000000,165.000000,65.000000,3.416667,0.041667,0.041667,0.041667,0.041667,5.000000,...,2.291667,4.708333,0.588889,0.041667,0.006250,0.958333,0.875000,1.041667,0.000000,0.000000
75%,40034.250000,55.000000,170.000000,75.000000,3.666667,0.043056,0.043056,0.041667,0.041667,5.416667,...,2.750000,5.666667,0.630556,0.041667,0.041667,1.208333,1.291667,1.833333,0.000000,1.000000
max,55690.000000,85.000000,190.000000,135.000000,5.375000,0.381250,0.381250,0.083333,0.083333,10.000000,...,25.750000,77.500000,0.875694,0.250000,0.462500,54.625000,121.416667,41.625000,1.000000,1.000000


In [6]:
# Cuento los valores de cada columna de texto
print("GENDER:")
print(train["gender"].value_counts())

print("\nORAL:")
print(train["oral"].value_counts())

print("\nTARTAR:")
print(train["tartar"].value_counts())


GENDER:
gender
M    31784
F    18216
Name: count, dtype: int64

ORAL:
oral
Y    50000
Name: count, dtype: int64

TARTAR:
tartar
Y    27789
N    22211
Name: count, dtype: int64


In [7]:
# Hay filas exactamente repetidas???
print("Filas duplicadas en train:", train.duplicated().sum())


Filas duplicadas en train: 0


## Conclusiones del Discovery

- El dataset de entrenamiento (`train`) tiene **50.000 filas y 27 columnas**. El de predicción (`test`) tiene **5.692 filas y 26 columnas**: le falta `smoking`, que es justamente el **target** que vamos a predecir.
- **No hay datos faltantes ni filas duplicadas** → los datos están limpios.
- El target `smoking` está **desbalanceado**: ~63% no fuma (0) y ~37% fuma (1).`M`
- Hay **3 columnas de texto**: `gender` (M/F), `oral` y `tartar` (Y/N). Habrá que convertirlas a número.
- **`oral` es constante** (siempre 'Y') → se descarta porque no aporta información.
- **`ID`** es solo un identificador → se descarta para entrenar (no es una característica real).
- Las columnas numéricas están en **escalas muy distintas** → habrá que escalarlas para modelos como KNN.
- Algunas columnas de hígado (`AST`, `ALT`, `Gtp`) tienen **valores extremos (outliers)**, algo normal en datos médicos.
